kolibri/content/api.py

```diff
@@ -307,6 +310,27 @@ def retrieve(self, request, pk):
 
         return response
 
+    @detail_route(methods=['get'])
+    def filesizes(self, request, pk):
+        queryset = self.get_queryset(prefetch=False)
+        instance = get_object_or_404(queryset, pk=pk)
+        descendants = instance.get_descendants(include_self=True).prefetch_related('files__local_file')
+
+        total_file_size = 0
+        on_device_file_size = 0
+        total_files = set()
+
+        for child in descendants:
+            for f in child.files.all():
+                total_files.add(f.local_file)
+
+        for local_file in total_files:
+            total_file_size += local_file.file_size or 0
+            if local_file.available:
+                on_device_file_size += local_file.file_size or 0
+
+        return Response({'total_file_size': total_file_size, 'on_device_file_size': on_device_file_size})
```

Comment:
It should be possible to replace most of the rest of this function with something like:

    files = LocalFile.objects.filter(files__contentnode__in=instance.get_descendants(include_self=True)).distinct()
    total_file_size = files.aggregate(Sum('file_size'))['file_size__sum'] or 0
    on_device_file_size = files.filter(available=True).aggregate(Sum('file_size'))['file_size__sum'] or 0

(you may need to add `from django.db.models import Sum` at the top)